arrays/arrays.c

```diff
@@ -45,14 +57,25 @@ void destroy_array(Array *arr) {
  * from old to new
  *****/
 void resize_array(Array *arr) {
-
   // Create a new element storage with double capacity
+  char **newElements = calloc((2 * arr->capacity), sizeof(char *)); // use calloc for contiguous memory
 
   // Copy elements into the new storage
+  for(int i = 0; i < arr->count; i++){ // use i < arr->count and not i <= arr->count since when i == arr->count, we're beyond valid indices
+    newElements[i] = arr->elements[i];
+  }
 
   // Free the old elements array (but NOT the strings they point to)
+  for(int i = 0; i < arr->count; i++){
```

Comment:
I think this for loop is what the comment means about freeing the strings the elements array points to (or NOT doing that to be precise). If your code works fine then I guess it's not strictly a problem, but your code should work without this for loop. Honestly I think there may be some slight misunderstandings/hand-wavings that went into the making of this project repo but at any rate you can do without this for loop.